core/src/test/java/org/apache/iceberg/jdbc/TestJdbcCatalog.java

```diff
@@ -1020,4 +1062,36 @@ private void initLegacySchema(String jdbcUrl) throws SQLException {
           .execute();
     }
   }
+
+  private boolean catalogTablesExist(String jdbcUrl) throws SQLException {
+    SQLiteDataSource dataSource = new SQLiteDataSource();
+    dataSource.setUrl(jdbcUrl);
+
+    boolean catalogTableExists = false;
+    boolean namespacePropertiesTableExists = false;
+
+    try (Connection connection = dataSource.getConnection()) {
+      DatabaseMetaData metadata = connection.getMetaData();
+      if (tableExists(metadata, JdbcUtil.CATALOG_TABLE_VIEW_NAME)) {
+        catalogTableExists = true;
+      }
+      if (tableExists(metadata, JdbcUtil.NAMESPACE_PROPERTIES_TABLE_NAME)) {
+        namespacePropertiesTableExists = true;
+      }
+    }
+
+    return catalogTableExists && namespacePropertiesTableExists;
+  }
+
+  boolean tableExists(DatabaseMetaData metadata, String tableName) throws SQLException {
+    ResultSet resultSet = metadata.getTables(null, null, tableName, new String[] {"TABLE"});
+
+    while (resultSet.next()) {
+      if (resultSet.getString("TABLE_NAME").equals(tableName)) {
```

Comment:
I would switch this around to `tableName.equals(resultSet.getString("TABLE_NAME"))` in case `resultSet.getString("TABLE_NAME")` returns `null`